src/db/schemas/user.ts

```diff
@@ -69,7 +70,7 @@ UserSchema.pre('save', function(next) {
 
 UserSchema.post('save', function() {
   const user = this;
-  sendMail(this.email, this.name);
+  sendMail(this.email, this.name, emailActions.CONFIRM_REGISTRATION);
```

Comment:
This is a potential pitfall.
You're missing the callback (please see) http://mongoosejs.com/docs/middleware.html
Also sendMail for 99% is async but .post will never know when it's ended.